In [1]:
import pandas as pd

In [11]:
df=pd.read_csv('random_forest_tt_scores.csv')
df1=pd.read_csv('random_forest_total_journey_tt_scores.csv')
df1=df1.rename(columns={"<60": "total<60", "<120": "total<120",">120": "total>120","overall": "total_overall"})

In [12]:
df=pd.merge(df, df1, on=['route'])

In [13]:
df['>120']=round(df['>120']/60, 1)
df['<120']=round(df['<120']/60, 1)
df['<60']=round(df['<60']/60, 1)
df['<30']=round(df['<30']/60, 1)
df['<10']=round(df['<10']/60, 1)
#df['<60']=round(df.sort_values('overall')['<60']/60, 2)
df['overall']=round(df['overall']/60, 1)
df['total<60']=round(df['total<60']/60, 1)
df['total<120']=round(df['total<120']/60, 1)
df['total>120']=round(df['total>120']/60, 1)
df['total_overall']=round(df['total_overall']/60, 1)

In [14]:
df.head()

route    <3  <10  <30   <60  <120  >120  overall  total<60  total<120  \
0   116  45.0  2.3  5.2  10.1  19.4   NaN      5.9       9.9       18.3   
1   118  58.0  1.6  2.7   8.3   NaN   NaN      2.4      11.4        6.6   
2    11  32.0  1.2  3.0  14.1  30.4  98.4      3.5       5.1       18.0   
3   120  24.0  1.0  2.5  14.8  57.5   NaN      1.1       3.9       11.8   
4   122  30.0  1.1  2.4   7.7  25.7   NaN      2.6       5.0        8.7   

   total>120  total_overall  
0       28.8           17.0  
1        NaN            7.4  
2       48.0           16.4  
3        NaN            4.1  
4       63.0            7.5

## Check if segments overall is within our window of error

In [15]:
df.loc[df.overall>5].sort_values('overall', ascending=False)

route    <3  <10  <30   <60  <120  >120  overall  total<60  total<120  \
89   68X  30.0  1.8  7.7  20.6  53.5   NaN      8.1      21.4       32.5   
0    116  45.0  2.3  5.2  10.1  19.4   NaN      5.9       9.9       18.3   
62   41X  69.0  3.4  5.6  16.9   NaN   NaN      5.2       4.0        9.1   

    total>120  total_overall  
89        NaN           29.4  
0        28.8           17.0  
62       39.2            9.4

In [16]:
df.loc[df.overall>3].sort_values('overall', ascending=False)

route    <3  <10  <30   <60  <120   >120  overall  total<60  total<120  \
89    68X  30.0  1.8  7.7  20.6  53.5    NaN      8.1      21.4       32.5   
0     116  45.0  2.3  5.2  10.1  19.4    NaN      5.9       9.9       18.3   
62    41X  69.0  3.4  5.6  16.9   NaN    NaN      5.2       4.0        9.1   
84    66X  99.0  3.1  4.7   8.5  20.9    NaN      5.0       6.8        8.3   
95    77X  28.0  1.4  3.5   NaN   NaN    NaN      4.8       NaN        8.3   
105    84  58.0  2.4  4.1   9.6  17.9    NaN      4.4       7.5        8.9   
33    27X  58.0  2.1  3.8   NaN   NaN    NaN      4.2      11.1       13.2   
67     44  50.0  1.8  3.8   7.9  12.9    NaN      4.1       5.8        9.2   
106   84X  36.0  1.7  3.6   6.5  25.2    NaN      4.1       5.8        9.0   
17    15D  60.0  2.5  5.0   8.7   NaN    NaN      4.1       9.4        9.8   
58    41B  60.0  2.5  5.0  12.2  38.0    NaN      4.0      12.1       12.8   
86    67X  59.0  1.8  3.8   6.9  11.0    NaN      4.0       8.0        8.7   
43     33  41.0  1.9  4.0   7.0  18.0   56.2      4.0       9.2        9.8   
51     39  41.0  1.8  3.8   7.2  21.2    NaN      3.8       7.2        9.6   
21     16  47.0  1.7  3.4   8.2  13.7    NaN      3.8       8.0        8.8   
28    25X  64.0  1.8  3.8   9.0   NaN    NaN      3.7      11.8        7.6   
60    41D  48.0  3.2  5.0   7.4  16.8    NaN      3.6       5.8       11.9   
73    51D  51.0  1.9  3.3   6.0   NaN    NaN      3.6       4.4        6.8   
2      11  32.0  1.2  3.0  14.1  30.4   98.4      3.5       5.1       18.0   
20    16D  50.0  1.9  3.8  14.1   NaN    NaN      3.5       NaN        7.7   
99     7B  51.0  1.7  3.5   6.1  17.5    NaN      3.5      11.8        9.5   
50    39A  36.0  1.4  3.2   7.1  17.0    NaN      3.4       6.0        7.8   
52    39X  56.0  1.6  3.3   9.8   NaN    NaN      3.4       8.9        8.0   
7      13  29.0  1.2  2.6   8.0  11.3    NaN      3.4       6.5        9.4   
94    77A  30.0  1.6  3.4   7.8  27.6  127.4      3.4       8.7       10.8   
92    70D  75.0  2.4  4.0  11.1   NaN    NaN      3.4      11.0        4.8   
40    32X  63.0  2.0  3.2   4.6   NaN    NaN      3.4       6.5        6.5   
47    38B  42.0  1.7  3.3   8.4  23.5    NaN      3.4       5.7       11.4   
32     27  31.0  1.3  2.6   7.0  16.8   75.6      3.4       5.7        8.8   
87    68A  79.0  2.6  5.6   NaN   NaN    NaN      3.3       6.5       15.7   
78     61  50.0  2.3  4.3   8.1  19.6  115.5      3.3       7.5       12.8   
18     15  41.0  1.6  3.0   9.7  19.6  151.2      3.3       8.0       11.4   
12     14  34.0  1.3  3.1   8.5  14.0  116.4      3.2       6.0        8.2   
9     142  33.0  1.4  2.9   6.4  10.0    NaN      3.2       2.8        8.8   
103    83  43.0  1.6  3.0   9.1  14.1    NaN      3.2       7.4        8.6   
56     40  37.0  1.4  2.7   7.8  12.6   86.1      3.2       6.0        8.2   
102   83A  33.0  1.3  3.0  10.4  21.2    NaN      3.1       8.8        8.6   
69    46E  45.0  1.7  3.3   7.9   NaN    NaN      3.1      14.9        7.7   

     total>120  total_overall  
89         NaN           29.4  
0         28.8           17.0  
62        39.2            9.4  
84        19.9            8.8  
95        13.3            8.6  
105       31.4            9.5  
33         NaN           12.8  
67        20.7            9.9  
106       27.7           10.0  
17         NaN            9.6  
58         NaN           12.3  
86        14.2            9.1  
43        28.9           11.1  
51        29.7           10.0  
21        23.7            9.2  
28        36.4            7.8  
60         NaN            7.2  
73        42.3            6.5  
2         48.0           16.4  
20        25.9            8.6  
99        30.9            9.8  
50        21.3            8.2  
52        33.1            8.3  
7         21.9           11.1  
94        38.0           11.0  
92         NaN            6.3  
40        28.7            6.8  
47        59.8           10.1  
32        18.2            9.8  
8

In [17]:
df.loc[df['<3']>60]

route    <3  <10  <30   <60  <120  >120  overall  total<60  total<120  \
28   25X  64.0  1.8  3.8   9.0   NaN   NaN      3.7      11.8        7.6   
40   32X  63.0  2.0  3.2   4.6   NaN   NaN      3.4       6.5        6.5   
55   40E  64.0  2.4  3.0  22.0   NaN   NaN      2.2       4.8        NaN   
62   41X  69.0  3.4  5.6  16.9   NaN   NaN      5.2       4.0        9.1   
74   51X  70.0  1.5  2.5   NaN   NaN   NaN      2.7       NaN        6.1   
84   66X  99.0  3.1  4.7   8.5  20.9   NaN      5.0       6.8        8.3   
87   68A  79.0  2.6  5.6   NaN   NaN   NaN      3.3       6.5       15.7   
92   70D  75.0  2.4  4.0  11.1   NaN   NaN      3.4      11.0        4.8   

    total>120  total_overall  
28       36.4            7.8  
40       28.7            6.8  
55        NaN            4.8  
62       39.2            9.4  
74        NaN            6.1  
84       19.9            8.8  
87        NaN            6.7  
92        NaN            6.3

In [19]:
list(df.loc[df['<3']>60]['route'].unique())

['25X', '32X', '40E', '41X', '51X', '66X', '68A', '70D']

## Check if these scores improve using all routes model

In [24]:
newscores=pd.read_csv('random_forest_all_routes_model_scores_per_route.csv')

In [25]:
newscores['>120']=round(newscores['>120']/60, 1)
newscores['<120']=round(newscores['<120']/60, 1)
newscores['<60']=round(newscores['<60']/60, 1)
newscores['<30']=round(newscores['<30']/60, 1)
newscores['<10']=round(newscores['<10']/60, 1)
#newscores['<60']=round(newscores.sort_values('overall')['<60']/60, 2)
newscores['overall']=round(newscores['overall']/60, 1)
newscores['total_<60']=round(newscores['total_<60']/60, 1)
newscores['total_<120']=round(newscores['total_<120']/60, 1)
newscores['total_>120']=round(newscores['total_>120']/60, 1)
newscores['total_overall']=round(newscores['total_overall']/60, 1)

In [28]:
newscores=pd.merge(df, newscores, on=['route'])

In [42]:
newscores.loc[newscores['<3_x']>newscores['<3_y']][['route','<3_x', '<3_y']]

route  <3_x  <3_y
13    150  36.0  33.0
14    151  39.0  37.0
16    15B  39.0  37.0
20    16D  50.0  37.0
21     16  47.0  44.0
23      1  41.0  39.0
26    25D  53.0  32.0
28    25X  64.0  62.0
29     26  45.0  31.0
33    27X  58.0  39.0
34    29A  39.0  34.0
40    32X  63.0  41.0
41    33D  48.0  37.0
43     33  41.0  40.0
45     37  37.0  31.0
48    38D  57.0  35.0
51     39  41.0  36.0
52    39X  56.0  38.0
53    40B  45.0  36.0
54    40D  37.0  31.0
55    40E  64.0  53.0
57    41A  42.0  35.0
58    41B  60.0  53.0
59    41C  56.0  44.0
61     41  54.0  52.0
62    41X  69.0  42.0
65     43  37.0  35.0
67     44  50.0  45.0
69    46E  45.0  39.0
71     49  41.0  33.0
74    51X  70.0  46.0
75     53  52.0  38.0
76    54A  37.0  34.0
77    56A  47.0  38.0
78     61  50.0  37.0
84    66X  99.0  51.0
86    67X  59.0  38.0
87    68A  79.0  41.0
90     69  40.0  31.0
92    70D  75.0  47.0
93     70  38.0  30.0
103    83  43.0  40.0
106   84X  36.0  29.0
107     9  53.0  46.0

In [36]:
pd.set_option('display.max_columns', 500)

In [40]:
newscores.loc[(newscores['<3_x']>newscores['<3_y'])&(newscores['overall_x']>newscores['overall_y'])]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
14    151  39.0    1.3    2.7    9.0    27.8     NaN        2.4       5.1   
16    15B  39.0    1.6    3.4   10.8    30.7   128.8        2.7       6.4   
21     16  47.0    1.7    3.4    8.2    13.7     NaN        3.8       8.0   
26    25D  53.0    1.6    3.5   11.2     NaN     NaN        3.0       7.4   
28    25X  64.0    1.8    3.8    9.0     NaN     NaN        3.7      11.8   
29     26  45.0    1.5    3.2   16.6     NaN     NaN        1.7       5.0   
33    27X  58.0    2.1    3.8    NaN     NaN     NaN        4.2      11.1   
40    32X  63.0    2.0    3.2    4.6     NaN     NaN        3.4       6.5   
43     33  41.0    1.9    4.0    7.0    18.0    56.2        4.0       9.2   
48    38D  57.0    1.8    3.4   10.9     NaN     NaN        2.5       5.3   
52    39X  56.0    1.6    3.3    9.8     NaN     NaN        3.4       8.9   
53    40B  45.0    1.5    2.8   11.0     NaN     NaN        1.9       5.1   
55    40E  64.0    2.4    3.0   22.0     NaN     NaN        2.2       4.8   
57    41A  42.0    1.4    2.8   10.8    26.1     NaN        2.6       5.5   
58    41B  60.0    2.5    5.0   12.2    38.0     NaN        4.0      12.1   
61     41  54.0    1.7    3.6    9.8    16.4   131.1        2.9       7.1   
62    41X  69.0    3.4    5.6   16.9     NaN     NaN        5.2       4.0   
67     44  50.0    1.8    3.8    7.9    12.9     NaN        4.1       5.8   
69    46E  45.0    1.7    3.3    7.9     NaN     NaN        3.1      14.9   
74    51X  70.0    1.5    2.5    NaN     NaN     NaN        2.7       NaN   
75     53  52.0    2.0    4.2   26.3    57.2     NaN        2.1       4.6   
76    54A  37.0    1.4    2.8    8.6    40.8     NaN        2.2       4.9   
77    56A  47.0    1.8    3.6    9.5    33.0     NaN        2.9       6.9   
78     61  50.0    2.3    4.3    8.1    19.6   115.5        3.3       7.5   
84    66X  99.0    3.1    4.7    8.5    20.9     NaN        5.0       6.8   
86    67X  59.0    1.8    3.8    6.9    11.0     NaN        4.0       8.0   
87    68A  79.0    2.6    5.6    NaN     NaN     NaN        3.3       6.5   
90     69  40.0    1.4    2.8    7.0    22.7     NaN        2.4       5.3   
92    70D  75.0    2.4    4.0   11.1     NaN     NaN        3.4      11.0   
93     70  38.0    1.6    3.3   11.9    36.6     NaN        3.0       6.5   
106   84X  36.0    1.7    3.6    6.5    25.2     NaN        4.1       5.8   

     total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
14         6.4      106.7              5.9  37.0    1.3    2.6    8.3    28.0   
16        12.1       59.2              8.3  37.0    1.5    3.3   10.6    31.4   
21         8.8       23.7              9.2  44.0    1.6    3.1    8.0    14.2   
26         6.4        NaN              7.0  32.0    1.1    2.4   10.9     NaN   
28         7.6       36.4              7.8  62.0    1.6    3.2   10.8     NaN   
29        32.7        NaN              5.4  31.0    1.1    3.1   16.9     NaN   
33        13.2        NaN             12.8  39.0    1.6    3.4    NaN     NaN   
40         6.5       28.7              6.8  41.0    1.5    2.6    4.6     NaN   
43         9.8       28.9             11.1  40.0    2.0    3.9    6.3    17.7   
48         7.4        NaN              6.1  35.0    1.1    3.7   12.5     NaN   
52         8.0       33.1              8.3  38.0    1.2    2.8    9.0     NaN   
53        17.4        NaN              5.6  36.0    1.3    2.6   11.8     NaN   
55         NaN        NaN              4.8  53.0    2.1    3.2   19.8     NaN   
57        10.6        NaN              8.9  35.0    1.4    2.6   11.0    27.9   
58        12.8        NaN             12.3  53.0    2.3    4.8   12.4    38.8   
61        10.1       96.6              8.7  52.0    1.6    3.2    9.3    17.8   
62         9.1       39.2              9.4  42.0    1.9    4.8   15.9     NaN   
67         9.2       20.7              9.9  45.0    1.6    3.4    7.2    13.2   


In [41]:
newscores.loc[(newscores['total_overall_x']>newscores['total_overall_y'])]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
0     116  45.0    2.3    5.2   10.1    19.4     NaN        5.9       9.9   
2      11  32.0    1.2    3.0   14.1    30.4    98.4        3.5       5.1   
3     120  24.0    1.0    2.5   14.8    57.5     NaN        1.1       3.9   
4     122  30.0    1.1    2.4    7.7    25.7     NaN        2.6       5.0   
5     123  26.0    1.0    2.3   10.4    35.2     NaN        2.2       4.4   
6     130  21.0    0.9    2.0   21.7    60.5     NaN        1.1       4.0   
7      13  29.0    1.2    2.6    8.0    11.3     NaN        3.4       6.5   
8     140  30.0    1.0    2.3   10.6    26.9     NaN        2.2       4.1   
9     142  33.0    1.4    2.9    6.4    10.0     NaN        3.2       2.8   
15    15A  38.0    1.7    3.2   10.2    38.8     NaN        2.3       5.5   
22    17A  37.0    1.2    2.7   10.5    19.2     NaN        2.5       5.2   
24    25A  31.0    1.2    2.7    7.6    20.2     NaN        2.1       5.5   
25    25B  32.0    1.3    2.6    7.7    32.8     NaN        2.1       5.3   
29     26  45.0    1.5    3.2   16.6     NaN     NaN        1.7       5.0   
33    27X  58.0    2.1    3.8    NaN     NaN     NaN        4.2      11.1   
38     31  24.0    1.0    2.2   12.2     NaN     NaN        1.5       4.6   
42    33E  20.0    1.1    2.3    NaN     NaN     NaN        2.8       NaN   
43     33  41.0    1.9    4.0    7.0    18.0    56.2        4.0       9.2   
47    38B  42.0    1.7    3.3    8.4    23.5     NaN        3.4       5.7   
57    41A  42.0    1.4    2.8   10.8    26.1     NaN        2.6       5.5   
58    41B  60.0    2.5    5.0   12.2    38.0     NaN        4.0      12.1   
61     41  54.0    1.7    3.6    9.8    16.4   131.1        2.9       7.1   
66    44B  32.0    1.5    4.3   19.3    57.4     NaN        1.8       5.4   
67     44  50.0    1.8    3.8    7.9    12.9     NaN        4.1       5.8   
73    51D  51.0    1.9    3.3    6.0     NaN     NaN        3.6       4.4   
77    56A  47.0    1.8    3.6    9.5    33.0     NaN        2.9       6.9   
78     61  50.0    2.3    4.3    8.1    19.6   115.5        3.3       7.5   
89    68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   
93     70  38.0    1.6    3.3   11.9    36.6     NaN        3.0       6.5   
97     79  28.0    1.2    2.6   11.8    52.7     NaN        1.6       4.6   
98     7A  42.0    1.5    2.9    5.9    21.6     NaN        2.6       6.8   
99     7B  51.0    1.7    3.5    6.1    17.5     NaN        3.5      11.8   
105    84  58.0    2.4    4.1    9.6    17.9     NaN        4.4       7.5   

     total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
0         18.3       28.8             17.0  47.0    2.1    4.4    7.3    17.4   
2         18.0       48.0             16.4  40.0    1.6    3.5   14.2    30.0   
3         11.8        NaN              4.1  30.0    1.1    2.4   14.7    56.6   
4          8.7       63.0              7.5  38.0    1.4    2.7    7.5    24.4   
5          9.2      112.3              6.7  35.0    1.3    2.6   10.4    34.0   
6         36.2        NaN              4.0  36.0    1.3    2.2   20.6    59.8   
7          9.4       21.9             11.1  39.0    1.6    3.2    8.2    10.3   
8         10.8        NaN              8.0  37.0    1.3    2.6   10.4    25.3   
9          8.8       36.1              5.9  41.0    1.5    3.0    6.5    10.5   
15        13.2        NaN              6.4  47.0    1.6    3.0   10.8    40.5   
22         7.9       44.5              7.2  43.0    1.3    2.6   10.1    18.6   
24         9.6        NaN              6.9  34.0    1.2    2.5    7.4    21.5   
25         8.7       54.2              6.8  36.0    1.2    2.4    7.5    32.2   
29        32.7        NaN              5.4  31.0    1.1    3.1   16.9     NaN   
33        13.2        NaN             12.8  39.0    1.6    3.4    NaN     NaN   
38        14.4        NaN              5.0  35.0    1.3    2.3   11.0     NaN   
42      

Although using the all routes model might be better over a particular segment for certain routes, it may not be more accurate overall. Therefore I will only use this model if a route doesn't meet our evaluations.

### These routes don't meet our under .3 of total journey

#### 3 Minutes

In [60]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['<3_x']>60]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
28   25X  64.0    1.8    3.8    9.0     NaN     NaN        3.7      11.8   
40   32X  63.0    2.0    3.2    4.6     NaN     NaN        3.4       6.5   
55   40E  64.0    2.4    3.0   22.0     NaN     NaN        2.2       4.8   
62   41X  69.0    3.4    5.6   16.9     NaN     NaN        5.2       4.0   
74   51X  70.0    1.5    2.5    NaN     NaN     NaN        2.7       NaN   
84   66X  99.0    3.1    4.7    8.5    20.9     NaN        5.0       6.8   
87   68A  79.0    2.6    5.6    NaN     NaN     NaN        3.3       6.5   
92   70D  75.0    2.4    4.0   11.1     NaN     NaN        3.4      11.0   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
28        7.6       36.4              7.8  62.0    1.6    3.2   10.8     NaN   
40        6.5       28.7              6.8  41.0    1.5    2.6    4.6     NaN   
55        NaN        NaN              4.8  53.0    2.1    3.2   19.8     NaN   
62        9.1       39.2              9.4  42.0    1.9    4.8   15.9     NaN   
74        6.1        NaN              6.1  46.0    1.2    2.4    NaN     NaN   
84        8.3       19.9              8.8  51.0    1.6    3.4    9.7    21.9   
87       15.7        NaN              6.7  41.0    1.9    6.2    NaN     NaN   
92        4.8        NaN              6.3  47.0    1.8    2.6    9.3     NaN   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
28     NaN        3.3        5.0         7.8        45.8              8.1  
40     NaN        3.1        7.7         7.3        30.7              7.6  
55     NaN        2.0        6.3         NaN         NaN              6.3  
62     NaN        5.1        1.9        12.5        51.8             12.8  
74     NaN        2.5        1.2         6.5         NaN              6.4  
84     NaN        3.9        6.4         9.3        28.8             10.2  
87     NaN        3.2        8.3        18.7         NaN              8.5  
92     NaN        2.5        6.0         6.9         NaN              6.6

I will use all routes model for these routes, as they fall within our criteria when using it.

In [45]:
all_routes_model_routes=list(newscores.loc[newscores['<3_x']>60]['route'].unique())

#### 10 Minutes

In [47]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['<10_x']>3.3]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
62   41X  69.0    3.4    5.6   16.9     NaN     NaN        5.2       4.0   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
62        9.1       39.2              9.4  42.0    1.9    4.8   15.9     NaN   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
62     NaN        5.1        1.9        12.5        51.8             12.8

Already flagged.

#### 30 Minutes

In [59]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['<30_x']>10]['route'].unique()))]

Empty DataFrame
Columns: [route, <3_x, <10_x, <30_x, <60_x, <120_x, >120_x, overall_x, total<60, total<120, total>120, total_overall_x, <3_y, <10_y, <30_y, <60_y, <120_y, >120_y, overall_y, total_<60, total_<120, total_>120, total_overall_y]
Index: []

#### 60 Minutes

In [50]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['<60_x']>20]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
6    130  21.0    0.9    2.0   21.7    60.5     NaN        1.1       4.0   
55   40E  64.0    2.4    3.0   22.0     NaN     NaN        2.2       4.8   
75    53  52.0    2.0    4.2   26.3    57.2     NaN        2.1       4.6   
89   68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
6        36.2        NaN              4.0  36.0    1.3    2.2   20.6    59.8   
55        NaN        NaN              4.8  53.0    2.1    3.2   19.8     NaN   
75       33.9        NaN              6.9  38.0    1.6    4.4   28.2    59.7   
89       32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
6      NaN        1.4        3.6        34.4         NaN              3.6  
55     NaN        2.0        6.3         NaN         NaN              6.3  
75     NaN        1.8        4.7        36.2         NaN              7.2  
89     NaN        6.9       20.2        30.5         NaN             27.7

I will add these routes to the list to use with all_routes_model as the majority of scores improve and fall closer to the evaluation cut-offs 

In [61]:
list(newscores.loc[newscores['<60_x']>20]['route'].unique())

['130', '40E', '53', '68X']

In [62]:
for route in list(newscores.loc[newscores['<60_x']>20]['route'].unique()):
    if route not in all_routes_model_routes:
        all_routes_model_routes+=route,

In [63]:
all_routes_model_routes

['25X', '32X', '40E', '41X', '51X', '66X', '68A', '70D', '130', '53', '68X']

#### 120 Minutes

In [65]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['<120_x']>40]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
3    120  24.0    1.0    2.5   14.8    57.5     NaN        1.1       3.9   
6    130  21.0    0.9    2.0   21.7    60.5     NaN        1.1       4.0   
13   150  36.0    1.2    2.8   11.0    50.4     NaN        1.6       4.7   
30   27A  30.0    1.2    2.1   15.1    46.3     NaN        1.3       3.6   
31   27B  29.0    1.3    2.3   10.7    42.0   110.4        1.6       4.2   
35   31A  38.0    1.4    2.4   15.2    41.2     NaN        2.1       4.6   
59   41C  56.0    1.8    3.3    8.8    46.1    99.5        2.8       6.4   
64    42  27.0    1.3    2.4    9.8    52.7     NaN        2.0       4.6   
66   44B  32.0    1.5    4.3   19.3    57.4     NaN        1.8       5.4   
71    49  41.0    1.5    2.8    9.0    42.0     NaN        2.2       5.1   
75    53  52.0    2.0    4.2   26.3    57.2     NaN        2.1       4.6   
76   54A  37.0    1.4    2.8    8.6    40.8     NaN        2.2       4.9   
89   68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   
97    79  28.0    1.2    2.6   11.8    52.7     NaN        1.6       4.6   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
3        11.8        NaN              4.1  30.0    1.1    2.4   14.7    56.6   
6        36.2        NaN              4.0  36.0    1.3    2.2   20.6    59.8   
13       18.4       95.7              5.2  33.0    1.3    2.7   11.4    51.4   
30       27.9        NaN              3.8  37.0    1.3    2.3   15.6    46.2   
31       11.6      115.8              4.9  32.0    1.3    2.4   10.8    41.2   
35       11.1        NaN              6.2  40.0    1.6    3.0   15.8    41.3   
59        8.7       78.1              7.8  44.0    1.6    3.5    9.1    47.4   
64        8.0       96.4              5.6  35.0    1.4    2.6   10.1    52.6   
66       61.7        NaN              5.5  35.0    1.4    3.8   18.1    60.4   
71        9.7       72.6              7.4  33.0    1.4    2.8   10.8    45.0   
75       33.9        NaN              6.9  38.0    1.6    4.4   28.2    59.7   
76        8.8       74.0              6.9  34.0    1.2    2.6    9.3    40.8   
89       32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   
97       29.8        NaN              5.1  29.0    1.1    2.5   11.9    53.0   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
3      NaN        1.2        3.4         9.4         NaN              3.6  
6      NaN        1.4        3.6        34.4         NaN              3.6  
13     NaN        1.6        4.8        19.4        97.4              5.3  
30     NaN        1.4        3.8        28.2         NaN              3.9  
31   111.4        1.7        4.4        11.0       114.6              5.1  
35     NaN        2.4        6.1        10.9         NaN              7.2  
59   101.0        2.8        6.2         9.1        78.6              7.9  
64     NaN        2.2        5.0         8.6        97.1              6.0  
66     NaN        1.6        4.6        62.0         NaN              4.7  
71     NaN        2.2        5.0        12.2        75.8              8.5  
75     NaN        1.8        4.7        36.2         NaN              7.2  
76     NaN        2.1        4.7         9.9        75.1              7.4  
89     NaN        6.9       20.2        30.5         NaN             27.7  
97     NaN        1.5        4.6        29.4         NaN              5.0

There is no significant improvement here so I will not add any of these new routes to my list.

#### Total Journey

In [70]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['total_overall_x']>30]['route'].unique()))]

Empty DataFrame
Columns: [route, <3_x, <10_x, <30_x, <60_x, <120_x, >120_x, overall_x, total<60, total<120, total>120, total_overall_x, <3_y, <10_y, <30_y, <60_y, <120_y, >120_y, overall_y, total_<60, total_<120, total_>120, total_overall_y]
Index: []

#### All our total journey errors are within the half hour mark we set

In [69]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['total_overall_x']>20]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
89   68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
89       32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
89     NaN        6.9       20.2        30.5         NaN             27.7

In [72]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['total_overall_x']>15]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
0    116  45.0    2.3    5.2   10.1    19.4     NaN        5.9       9.9   
2     11  32.0    1.2    3.0   14.1    30.4    98.4        3.5       5.1   
89   68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
0        18.3       28.8             17.0  47.0    2.1    4.4    7.3    17.4   
2        18.0       48.0             16.4  40.0    1.6    3.5   14.2    30.0   
89       32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
0      NaN        4.5        5.9        14.0        27.8             12.8  
2     96.6        3.8        5.8        16.2        45.3             15.0  
89     NaN        6.9       20.2        30.5         NaN             27.7

The scores for 116 are significant enough to include them in our all_routes_model. Although the 11 has some scores better in their individual route model, all the scores are within out criteria when using the all_routes_model and so I will add it.

In [74]:
all_routes_model_routes+='116',
all_routes_model_routes+='11',

In [77]:
all_routes_model_routes

['25X',
 '32X',
 '40E',
 '41X',
 '51X',
 '66X',
 '68A',
 '70D',
 '130',
 '53',
 '68X',
 '116',
 '11']

In [76]:
newscores.loc[newscores.route.isin(list(newscores.loc[newscores['total_overall_x']>10]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
0     116  45.0    2.3    5.2   10.1    19.4     NaN        5.9       9.9   
2      11  32.0    1.2    3.0   14.1    30.4    98.4        3.5       5.1   
7      13  29.0    1.2    2.6    8.0    11.3     NaN        3.4       6.5   
18     15  41.0    1.6    3.0    9.7    19.6   151.2        3.3       8.0   
33    27X  58.0    2.1    3.8    NaN     NaN     NaN        4.2      11.1   
43     33  41.0    1.9    4.0    7.0    18.0    56.2        4.0       9.2   
47    38B  42.0    1.7    3.3    8.4    23.5     NaN        3.4       5.7   
58    41B  60.0    2.5    5.0   12.2    38.0     NaN        4.0      12.1   
79    65B  30.0    1.5    3.0   11.6    36.2     NaN        2.6       5.5   
80     65  31.0    1.4    3.1    7.0    22.6     NaN        2.9       7.9   
89    68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   
93     70  38.0    1.6    3.3   11.9    36.6     NaN        3.0       6.5   
94    77A  30.0    1.6    3.4    7.8    27.6   127.4        3.4       8.7   
101     7  40.0    1.5    2.9    8.2    29.7     NaN        2.5       6.4   

     total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
0         18.3       28.8             17.0  47.0    2.1    4.4    7.3    17.4   
2         18.0       48.0             16.4  40.0    1.6    3.5   14.2    30.0   
7          9.4       21.9             11.1  39.0    1.6    3.2    8.2    10.3   
18        11.4       28.8             11.5  51.0    1.7    3.2    9.9    20.3   
33        13.2        NaN             12.8  39.0    1.6    3.4    NaN     NaN   
43         9.8       28.9             11.1  40.0    2.0    3.9    6.3    17.7   
47        11.4       59.8             10.1  44.0    1.6    3.0    6.8    18.3   
58        12.8        NaN             12.3  53.0    2.3    4.8   12.4    38.8   
79        12.4       69.3             10.6  36.0    1.4    2.9   12.4    38.3   
80         9.4       31.7             10.7  33.0    1.5    3.1    7.2    23.4   
89        32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   
93        11.6       52.5             10.3  30.0    1.4    2.9   11.2    34.6   
94        10.8       38.0             11.0  33.0    1.5    3.2    7.3    26.2   
101       11.9       69.9             10.7  49.0    1.6    2.9    8.3    29.3   

     >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
0       NaN        4.5        5.9        14.0        27.8             12.8  
2      96.6        3.8        5.8        16.2        45.3             15.0  
7       NaN        3.8        6.8         8.6        17.0              9.8  
18    153.1        3.6        7.8        12.0        29.6             12.0  
33      NaN        3.7        8.6        12.7         NaN             12.2  
43     55.4        3.8        8.5         8.9        31.6             10.4  
47      NaN        2.9        7.0         9.3        50.0              8.8  
58      NaN        3.8       10.4        14.5         NaN             11.6  
79      NaN        2.5        5.4        14.3        72.0             11.7  
80      NaN        3.0        7.7         9.8        31.2             11.1  
89      NaN        6.9       20.2        30.5         NaN             27.7  
93      NaN        2.7        6.2        11.1        53.4              9.9  
94    124.6        3.3        8.1        10.9        37.4             11.0  
101     NaN        2.6        6.4        11.9        69.6             10.7

In [78]:
newscores.loc[newscores.route.isin(list(newscores.loc[(newscores['total_overall_x']>10)&(newscores['total_overall_x']>newscores['total_overall_y'])]['route'].unique()))]

route  <3_x  <10_x  <30_x  <60_x  <120_x  >120_x  overall_x  total<60  \
0    116  45.0    2.3    5.2   10.1    19.4     NaN        5.9       9.9   
2     11  32.0    1.2    3.0   14.1    30.4    98.4        3.5       5.1   
7     13  29.0    1.2    2.6    8.0    11.3     NaN        3.4       6.5   
33   27X  58.0    2.1    3.8    NaN     NaN     NaN        4.2      11.1   
43    33  41.0    1.9    4.0    7.0    18.0    56.2        4.0       9.2   
47   38B  42.0    1.7    3.3    8.4    23.5     NaN        3.4       5.7   
58   41B  60.0    2.5    5.0   12.2    38.0     NaN        4.0      12.1   
89   68X  30.0    1.8    7.7   20.6    53.5     NaN        8.1      21.4   
93    70  38.0    1.6    3.3   11.9    36.6     NaN        3.0       6.5   

    total<120  total>120  total_overall_x  <3_y  <10_y  <30_y  <60_y  <120_y  \
0        18.3       28.8             17.0  47.0    2.1    4.4    7.3    17.4   
2        18.0       48.0             16.4  40.0    1.6    3.5   14.2    30.0   
7         9.4       21.9             11.1  39.0    1.6    3.2    8.2    10.3   
33       13.2        NaN             12.8  39.0    1.6    3.4    NaN     NaN   
43        9.8       28.9             11.1  40.0    2.0    3.9    6.3    17.7   
47       11.4       59.8             10.1  44.0    1.6    3.0    6.8    18.3   
58       12.8        NaN             12.3  53.0    2.3    4.8   12.4    38.8   
89       32.5        NaN             29.4  36.0    1.7    6.2   17.8    51.4   
93       11.6       52.5             10.3  30.0    1.4    2.9   11.2    34.6   

    >120_y  overall_y  total_<60  total_<120  total_>120  total_overall_y  
0      NaN        4.5        5.9        14.0        27.8             12.8  
2     96.6        3.8        5.8        16.2        45.3             15.0  
7      NaN        3.8        6.8         8.6        17.0              9.8  
33     NaN        3.7        8.6        12.7         NaN             12.2  
43    55.4        3.8        8.5         8.9        31.6             10.4  
47     NaN        2.9        7.0         9.3        50.0              8.8  
58     NaN        3.8       10.4        14.5         NaN             11.6  
89     NaN        6.9       20.2        30.5         NaN             27.7  
93     NaN        2.7        6.2        11.1        53.4              9.9

The scores for the 38B and 70 are significant enough to add them to our all_routes_model

In [79]:
all_routes_model_routes+='38B',
all_routes_model_routes+='70',

### These are the routes that will use the all_routes_model.

In [80]:
all_routes_model_routes

['25X',
 '32X',
 '40E',
 '41X',
 '51X',
 '66X',
 '68A',
 '70D',
 '130',
 '53',
 '68X',
 '116',
 '11',
 '38B',
 '70']

# Results

- There is only one route that falls outside our 60 second error window for journeys under 3 minutes.
<br> 25X has a 62 sec error.
- There are no routes that fall outside our 3 min 20 sec error window for 10 minute journeys.
- There are no routes that fall outside our 10 min error window for 30 minute journeys.
- There are two routes that fall outside our 20 min error window for 60 minute journeys.
<br>The 130 has a 20 mins 10 sec error.
<br>The 53 has a 28 minutes error. This is a new route which wouldn't have been accounted for in our training of the model.
- There are no journeys that fall outside our 30 minutes total journey window. There is only one route that is outside a 15 minute window and 7 outside a 10 minute window.